src/harness/security_testcase.py

```diff
@@ -102,24 +103,83 @@ def _InfoCb(conn, where, ok):
       return ok
     ctx.set_info_callback(_InfoCb)
 
-    # only for potential debugging info...
-    def _VerifyCb(conn, cert, errnum, depth, ok):
-      certsubject = crypto.X509Name(cert.get_subject())
-      commonname = certsubject.commonName
-      logging.debug('TLS handshake verify: certificate: %s  -> %d', commonname, ok)
-      return ok
-    ctx.set_verify(SSL.VERIFY_PEER, _VerifyCb)
+    if IsVerifyServercert:
+      default_values = json.load(
+                      open(os.path.join('testcases', 'configs', 'test_WINNF_FT_S_SCS_20', 'default.config')))
+
+
+    def _ValidateSeverCertificateCb(conn, cert, errnum, depth, ok):
+        certsubject = crypto.X509Name(cert.get_subject())
+        commonname = certsubject.commonName
+        logging.debug('TLS handshake verify: certificate: %s  -> %d', commonname, ok)
+        if(default_values["Subject"] == crypto.X509Name(cert.get_subject()).commonName ):
+          cnotBefore = cert.get_notBefore()
+          cnotAfter = cert.get_notAfter()
+          pubkey = cert.get_pubkey()
+          serial_number = cert.get_serial_number()
+          certsubject = crypto.X509Name(cert.get_issuer())
+          commonname = certsubject.commonName
+          # only for potential debugging info...
+          logging.debug('serial number %s ', cert.get_serial_number())
+          logging.debug('signature algorithm %s ', cert.get_signature_algorithm())
+          logging.debug('Number of bits in Publick key  %s ', pubkey.bits())
+          # Checking the validity of the date of the certificate is taken care by the pyopenssl v17.5.0 library
+          self.assertIsNotNone(cnotAfter)
+          logging.debug('Not After %s ', cnotAfter)
+          self.assertIsNotNone(cnotBefore)
+          logging.debug('Not Before %s ', cnotBefore)
+          logging.debug('Common Name ->subject %s ', crypto.X509Name(cert.get_subject()).commonName)
+          logging.debug('Common Name ->Issuer %s ', crypto.X509Name(cert.get_issuer()).commonName)
+          logging.debug('organizationName %s ', crypto.X509Name(cert.get_issuer()).organizationName )
+          logging.debug('ROLE  %s ', cert.get_extension(4))
+          logging.debug('CA  %s ', cert.get_extension(2))
+          self.assertEqual(cert.get_issuer().commonName, default_values["Issuer"])
+          self.assertEqual(cert.get_subject().commonName, default_values["Subject"])
```

Comment:
Won't this check always pass because the if() condition above requires it?

Related: if the if() condition fails, doesn't this skip all of the certificate checks? Is that really the right behavior?

Reply:
Yes,it   will always pass as "if()" condition above  requires it and will skip all the   certificate checks if the "if()" condition fails.

Reply:
This function is invoked only for test case SCS.20. This function was separated out based on an earlier comment. Other test cases do not require validation of each field. 

Added additional comment on top of previous resolution for better understanding

Reply:
Even if it is invoked only for a single test case, we should still make sure the logic is correct. Right now, an incorrectly-implemented SAS could pass this test case by making sure that the if() condition fails, which we definitely don't want. Please ensure that this code is correctly executing the checks required by SCS.20.